syntax = "proto3";

service ImageGenService {
  rpc GenerateImage (ImageRequest) returns (ImageResponse);
}

message ImageRequest {
  string prompt = 1;
  int32 steps = 2;
}

message ImageResponse {
  string status = 1;
  string image_base64 = 2;
}
